src/workloads/query/BooleanSimplifier.yml

```diff
@@ -0,0 +1,208 @@
+        Database: *Database
+        MultipleThreadsPerCollection: true
+        CollectionCount: 1
+        DocumentCount: *DocumentCount
+        BatchSize: 1000
+        Document:
+          a1: {^RandomInt: {min: 0, max: 100}}
+          b1: {^Inc: {start: 11, step: 2}}
+          a2: {^RandomInt: {min: 0, max: 1000}}
+          b2: 3
+          c2: {^RandomInt: {min: 0, max: 5}}
+          d2: {^RandomInt: {min: 0, max: 5}}
+          e2: 4
+          a3: {^RandomInt: {min: 0, max: 100}}
+          b3: {^RandomInt: {min: 0, max: 100}}
+          c3: {^Array: {of: {^FastRandomString: {length: 11}}, number: 10}}
+        Indexes:
+        - keys: {b1: 1}
+        - keys: {c2: 1, d2: 1}
+
+# Ensure all data is synced to disk.
+- Name: Quiesce
+  Type: QuiesceActor
+  Threads: 1
+  Database: *Database
+  Phases:
+    OnlyActiveInPhases:
+      Active: [2]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Threads: 1
+
+# The simplifier optimizes out an $or's branch and makes an Index Scan plan possible instead of 
+# CollectionScan one
+- Name: SimplifiedOutOrBranch.BM
+  Type: CrudActor
+  Database: *Database
+  Threads: *Threads
+  Phases:
+    OnlyActiveInPhases:
+      Active: [3]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Duration: *Duration
+        Collection: Collection0
+        Operations:
+        - OperationName: find
+          OperationCommand:
+            Filter:  {$or: [{$and: [{a1: 1}, {a1: {$ne: 1}}]}, {b1: 2}]}
+
+# The simplifier factorizes out common terms and makes a plan with one fetch possible
+# instead of a plan with two fetches.
+- Name: SimplifierCollapsesTwoFetchesIntoOne.BM
+  Type: CrudActor
+  Database: *Database
+  Threads: *Threads
+  Phases:
+    OnlyActiveInPhases:
+      Active: [4]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Duration: *Duration
+        Collection: Collection0
+        Operations:
+        - OperationName: find
+          OperationCommand:
+            Filter:  {a2: 1, $or: [{c2: 1, d2: 2, b2: 3}, {c2: 1, d2: 2, e2: 4}]}
+
+- Name: SimplifiedCollScanFilter.BM
+  Type: CrudActor
+  Database: *Database
+  Threads: *Threads
+  Phases:
+    OnlyActiveInPhases:
+      Active: [5]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Duration: *Duration
+        Collection: Collection0
+        Operations:
+        - OperationName: find
+          OperationCommand:
+            Filter:  {
+                $and: [
+                    {b3: 0},
+                    {
+                        $or: [
+                            {a3: 0},
+                            {
+                                $and: [
+                                    {c3: {$in: *QueryStringIn}},
+                                    {c3: {$nin: *QueryStringIn}},
+                                ]
+                            }
+                        ]
+                    },
+                ]
+            }
+
+
+- Name: TrivialOr.BM
```

Comment:
I don't think we need the .BM suffix- IMO, genny metrics names are quite verbose enough